src/DataTypes/DataTypesNumber.cpp

```diff
@@ -2,23 +2,128 @@
 #include <DataTypes/DataTypeFactory.h>
 
 
+#include <Parsers/IAST.h>
+#include <Parsers/ASTLiteral.h>
+
+
 namespace DB
 {
 
+namespace ErrorCodes
+{
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+    extern const int UNEXPECTED_AST_STRUCTURE;
+}
+
+static DataTypePtr createForInt8(const ASTPtr & arguments)
+{
+    if (arguments) {
+        if (arguments->children.size() > 1)
+            throw Exception("INT8 data type family must not have more than one argument - display width", ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+
+        const auto * argument = arguments->children[0]->as<ASTLiteral>();
+        if (!argument || argument->value.getType() != Field::Types::UInt64 || argument->value.get<UInt64>() == 0)
```

Comment:
It seems excessive to do a lot of checks about arguments which you are even not going to use.
I think checking the number of arguments is enough here.

Reply:
but users cant pass -1 to string length. Or its ok?

Reply:
If user pass -1 we don't care, it can damage nothing because we ignore that value anyway.

Reply:
ok

Reply:
> If user pass -1 we don't care, it can damage nothing because we ignore that value anyway.

Sorry, I disagree.
It's better to check arguments.

Because later we can treat them as constraints or add some other meanings, but if we would not check argument, we will have to deal with VARCHAR(-1) and similar.

Reply:
Ok, then we should check both boundaries, should we?
I mean `TINYINT(1000000)` also seems incorrect.

Reply:
It seems MySQL allows `d` inside `TINYINT(d)` in range `0..255`. This seems to be a good range, so let's check that the `argument` for integral types is in range `1..255`.

Reply:
ok, what about float types?

Reply:
MySQL allows `d` inside `FLOAT(d, p)` and `DOUBLE(d, p)` in range `0..255` and `p` in range `0..30`, and `d` should be greater or equal to `p`: `d>=p`. Because of `Float64` has 15-17 significant digits it should be enough.

And for format `FLOAT(p) ` which we could also support it seems logical to check that p in range `1..53` (because `p` is precision in binary digits).